decoder.rb

```diff
@@ -0,0 +1,35 @@
+# decode morse code character
+def decode_char(character)
+  # dictionary
+  dictionary = {
+    'A' => '.-', 'B' => '-...', 'C' => '-.-.', 'D' => '-..', 'E' => '.', 'F' => '..-.',
+    'G' => '--.', 'H' => '....', 'I' => '..', 'J' => '.---', 'K' => '-.-', 'L' => '.-..',
+    'M' => '--', 'N' => '-.', 'O' => '---', 'P' => '.--.', 'Q' => '--.-', 'R' => '.-.', 'S' => '...',
+    'T' => '-', 'U' => '..-', 'V' => '...-', 'W' => '.--', 'X' => '-..-', 'Y' => '-.--', 'Z' => '--..'
+  }
+
+  dictionary.each do |key, value|
+    return key if character == value
+  end
+end
+
+def decode_word(word)
+  split_word = word.split
+  split_word.each do |i|
+    decode_char(i)
+  end
+end
```

Comment:
- [ ] This method returns an array of characters. Kindly fix that.

  ![didd](https://user-images.githubusercontent.com/81468675/207627574-f2b8627c-d26b-4931-9e11-156ca72bd6a5.jpg)

  Hint:
  ```
  def decode_word(word)
    split_word = word.split
    word = []
    split_word.each do |i|
      word << decode_char(i)
    end
    word.join
  end
  ```

Reply:
Thanks for the hint. The code has been refactored